File: react-type-vite/src/services/api/response/publishedCourseResponse.ts
import type { CourseResponse } from './courseResponse';
import type { CourseCategoryResponse } from './courseTypeResponse';
import type { SectionResponse } from './sectionResponse';


export interface PublishedCourseResponse {
  id: number;
  course: CourseResponse;
  courseType: CourseCategoryResponse;
  description: string;
  courseIntroduction: string;
  courseImage: string;
  courseVideo: string;
  learnerAchievements: string;
  courseLearner: string;
  courseTarget: string[];
  coursePrice: number;
  status: number; // 0: Draft, 1: Pending Approval, 2: Approved, 3: Rejected
  statusText: string;
  createdAt: string;
  updatedAt: string;
  publishedSections?: SectionResponse[]; // Sections với chỉ content đã published
  totalPublishedLessons?: number;
  totalPublishedQuizzes?: number;
  totalPublishedAssignments?: number;
}

File: spring-microservices/course-service/src/main/java/com/hoangphihiep/dto/response/PublishedCourseResponse.java
package com.hoangphihiep.dto.response;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.util.Date;
import java.util.List;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class PublishedCourseResponse {

    private Integer id;

    private CourseResponse course;

    private CourseTypeResponse courseType;

    private String description;

    private String courseIntroduction;

    private String courseImage;

    private String courseVideo;

    private String learnerAchievements;

    private String courseLearner;

    private List<String> courseTarget;

    private BigDecimal coursePrice;

    private Integer status; // 0: Draft, 1: Pending Approval, 2: Approved, 3: Rejected

    private String statusText;

    private Date createdAt;

    private Date updatedAt;

    private List<SectionResponse> publishedSections; // Các section được public

    private Integer totalPublishedLessons;

    private Integer totalPublishedQuizzes;

    private Integer totalPublishedAssignments;
}